### PR TITLE
Blocks by number

### PR DESCRIPTION
**Pull request by ajlopez**
New Web3 method

### CODE REVIEW

File: rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java [hunk 1]
```diff
@@ -686,6 +695,28 @@ public BlockResult getBlockResult(Block b, boolean fullTx) {
...
+        try {
+            blockNumber = StringHexToBigInteger(number).longValue();
+        } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
+            throw new JsonRpcInvalidParamException("invalid blocknumber " + number);
+        }
+
+        List<BlockInformationResult> result = new ArrayList<>();
+        Blockchain blockchain = this.worldManager.getBlockchain();
+
+        synchronized (blockchain) {
```
Comment (donequis) [inline rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java]: Do not sync over the blockchain object. Check accessLock object in blockchainImpl

**Comment by ajlopez** [comment]
The original BlockInfo is used with Java serialization by the current index store, so I created the new one to don't alter in any way the current behavior, and have the new method exposing data that is not related only with IndexedBlockStore

File: rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java [hunk 2]
```diff
@@ -686,6 +695,26 @@ public BlockResult getBlockResult(Block b, boolean fullTx) {
         return br;
     }
 
+    public BlockInformationResult[] eth_getBlocksByNumber(String number) {
+        long blockNumber;
+
+        try {
+            blockNumber = StringHexToBigInteger(number).longValue();
```
Comment (donequis) [inline rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java]: Consider the method StringNumberAsBigInt() instead of StringHexToBigInteger()

**Comment by donequis** [comment]
Create PR against ginger

**Pull request merged**